### PR TITLE
Improve post type labeling for v5

### LINKED ISSUE #13: Improve post type labeling
For example, if two post types share the same label name it becomes difficult to differentiate the two without looking at the values of each post type.

I suggest using `acf_get_pretty_post_types()`, which is what ACF uses for its relational field types (Post Object, Page Link, Relationship). The function handles label conflicts by appending the post type key to the label.

### PR DESCRIPTION
**Pull request by mcaskill**
~This code is untested yet, I only had enough time to prepare the labeling.~

> This code has been tested.

Replaced `get_post_types()` with `acf_get_pretty_post_types()` in field for v5.

Resolves #13

### CODE REVIEW

File: post-type-selector-v5.php [hunk 1]
```diff
@@ -151,11 +151,11 @@ function render_field( $field ) {
 
 				$checked[ $field[ 'value' ] ] = 'checked="checked"';
 
-				foreach( $post_types as $post_type ) {
+				foreach( $post_types as $post_type => $post_type_label ) {
 
 				?>
 
-					<li><input type="radio" <?php echo ( isset( $checked[ $post_type->name ] ) ) ? $checked[ $post_type->name] : null; ?> class="<?php echo $field[ 'class' ]; ?>" name="<?php echo $field[ 'name' ]; ?>" value="<?php echo $post_type->name; ?>"><label><?php echo $post_type->labels->name; ?></label></li>
+					<li><input type="radio" <?php echo ( isset( $checked[ $post_type ] ) ) ? $checked[ $post_type] : null; ?> class="<?php echo $field[ 'class' ]; ?>" name="<?php echo $field[ 'name' ]; ?>" value="<?php echo $post_type; ?>"><label><?php echo $post_type_label; ?></label></li>
```
Comment (tyrann0us) [inline post-type-selector-v5.php]: `$post_type_label` is an object of class `stdClass`; `<?php echo $post_type_label; ?>` will cause a recoverable fatal error.

File: post-type-selector-v5.php [hunk 2]
```diff
@@ -137,8 +137,8 @@ function render_field( $field ) {
 
 				$checked[ $field[ 'value' ] ] = 'selected="selected"';
 
-				foreach( $post_types as $post_type ) {
-					echo '<option ' . (isset($checked[ $post_type->name ]) ? $checked [ $post_type->name ] : null) . ' value="' . $post_type->name . '">' . $post_type->labels->name . '</option>';
+				foreach( $post_types as $post_type => $post_type_label ) {
+					echo '<option ' . (isset($checked[ $post_type ]) ? $checked [ $post_type ] : null) . ' value="' . $post_type . '">' . $post_type_label . '</option>';
```
Comment (tyrann0us) [inline post-type-selector-v5.php]: `$post_type_label` is an object of class `stdClass`; `<?php echo $post_type_label; ?>` will cause a recoverable fatal error.

File: post-type-selector-v5.php [hunk 3]
```diff
@@ -180,11 +180,11 @@ function render_field( $field ) {
 
 				}
 
-				foreach( $post_types as $post_type ) {
+				foreach( $post_types as $post_type => $post_type_label ) {
 
 				?>
 
-					<li><input type="checkbox" <?php echo ( isset( $checked[ $post_type->name ] ) ) ? $checked[ $post_type->name] : null; ?> class="<?php echo $field[ 'class' ]; ?>" name="<?php echo $field[ 'name' ]; ?>[]" value="<?php echo $post_type->name; ?>"><label><?php echo $post_type->labels->name; ?></label></li>
+					<li><input type="checkbox" <?php echo ( isset( $checked[ $post_type ] ) ) ? $checked[ $post_type] : null; ?> class="<?php echo $field[ 'class' ]; ?>" name="<?php echo $field[ 'name' ]; ?>[]" value="<?php echo $post_type; ?>"><label><?php echo $post_type_label; ?></label></li>
```
Comment (tyrann0us) [inline post-type-selector-v5.php]: `$post_type_label` is an object of class `stdClass`; `<?php echo $post_type_label; ?>` will cause a recoverable fatal error.

File: post-type-selector-v5.php (hunk 2 above)
Comment (mcaskill) [reply post-type-selector-v5.php]: Thanks for the comments. I'll try to look into this tonight with a test environment.

Comment (mcaskill) [reply post-type-selector-v5.php]: I fixed the `stdClass` issue which was caused by the "None" choice.

File: post-type-selector-v5.php (hunk 1 above)
Comment (mcaskill) [reply post-type-selector-v5.php]: I fixed the stdClass issue which was caused by the "None" choice.

File: post-type-selector-v5.php (hunk 3 above)
Comment (mcaskill) [reply post-type-selector-v5.php]: I fixed the stdClass issue which was caused by the "None" choice.

File: post-type-selector-v5.php [hunk 4]
```diff
@@ -168,6 +160,8 @@ function render_field( $field ) {
 
 			case acf_field_post_type_selector::SELECTOR_TYPE_CHECKBOXES:
 
+				unset( $post_types[ '' ] );
```
Comment (tyrann0us) [inline post-type-selector-v5.php]: What's the reason to remove the "None" checkbox? There seems to be a bug in the original code that prevents you from unsetting all checkboxes. The "None" checkbox is the only way to work around this.

Comment (mcaskill) [reply post-type-selector-v5.php]: Thanks for the heads up; I thought that might have been there for a reason.
I did not notice any saving issues with the latest ACF/WP.
I'll do some extra tests soon.

---

I have an upcoming pull request that updates the ACF 5 field to take advantage of functions like [`acf_get_select_input()`](https://github.com/AdvancedCustomFields/acf/blob/master/includes/api/api-input.php#L395) which should mitigates exceptions like those checkboxes.
I need to lookup ACF 4 for any equivalents.

**Comment by tyrann0us** [comment]
@mcaskill, I'm in favor to wait here until your `acf_get_select_input()` PR is merged. Thoughts?

**Comment by mcaskill** [comment]
> I'm in favor to wait here until your acf_get_select_input() PR is merged. Thoughts?

Sure! We can take this PR to the next step.

I was also considering a couple of other things:
- Deleting `css/`, `js/, and `cpt_field.php` which are unused;
- Adding `.editorconfig` to maintain a consistent style (consistent indentation, trim whitespace, etc.)

**Comment by hacknug** [comment]
Is this going to get merged soon? I'd like to open a PR moving the `input[type="radio"]` inside the label so it's easier to select them.

Right now you must click on the input itself because the input has no `ID` and the label is not linked to the input using the `for` attribute. Moving the input inside the label would be the easier way to deal with this issue since we wouldn't need to add id's to each one of those options.

**Comment by tyrann0us** [comment]
@mcaskill, I would like to "hand over" this PR to you. Right now, I barely find time to work on open source projects like this. As far as I can see, @TimPerry made you a collaborator so you should be able to merge this PR on your own.
Thanks!

**Comment by mcaskill** [comment]
Hi @hacknug,

I've also had trouble finding time to work on this plugin. I started work again this past weekend, cleaning up my PR and evaluating what else will need cleaning up based on the current ACF architecture. I might be able to merge this PR during this week.

**Pull request merged**